rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java

```diff
@@ -18,134 +18,100 @@
-            }
-
-            TransactionPoolAddResult result = transactionGateway.receiveTransaction(tx);
-            if(!result.transactionsWereAdded()) {
-                throw RskJsonRpcRequestException.transactionError(result.getErrorMessage());
-            }
-
-            return s = tx.getHash().toJsonString();
-        } finally {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("eth_sendRawTransaction({}): {}", rawData, s);
-            }
-        }
-    }
-
-    private static byte hexToChainId(String hex) {
-        if (hex == null) {
-            return 0;
-        }
-        try {
-            byte[] bytes = TypeConverter.stringHexToByteArray(hex);
-            if (bytes.length != 1) {
-                throw RskJsonRpcRequestException.invalidParamError(ERR_INVALID_CHAIN_ID + hex);
-            }
-
-            return bytes[0];
-        } catch (Exception e) {
-            throw RskJsonRpcRequestException.invalidParamError(ERR_INVALID_CHAIN_ID + hex, e);
-        }
-    }
+	protected static final Logger LOGGER = LoggerFactory.getLogger("web3");
+
+	private final Wallet wallet;
+	private final TransactionPool transactionPool;
+	private final Constants constants;
+	private final TransactionGateway transactionGateway;
+
+	public EthModuleTransactionBase(Constants constants, Wallet wallet, TransactionPool transactionPool, TransactionGateway transactionGateway) {
+		this.wallet = wallet;
+		this.transactionPool = transactionPool;
+		this.constants = constants;
+		this.transactionGateway = transactionGateway;
+	}
+
+	@Override
+	public synchronized String sendTransaction(Web3.CallArguments args) {
+
+		Account senderAccount = this.wallet.getAccount(new RskAddress(args.from));
+		String txHash = null;
+
+		try {
+
+			synchronized (transactionPool) {
+
+				TransactionArguments txArgs = TransactionArgumentsUtil.processArguments(args, transactionPool, senderAccount, constants.getChainId());
+
+				Transaction tx = Transaction.builder().from(txArgs).build();
+
+				tx.sign(senderAccount.getEcKey().getPrivKeyBytes());
+
+				if (!tx.acceptTransactionSignature(constants.getChainId())) {
+					throw RskJsonRpcRequestException.invalidParamError(TransactionArgumentsUtil.ERR_INVALID_CHAIN_ID + args.chainId);
+				}
+
+				TransactionPoolAddResult result = transactionGateway.receiveTransaction(tx.toImmutableTransaction());
+
+				if (!result.transactionsWereAdded()) {
+					throw RskJsonRpcRequestException.transactionError(result.getErrorMessage());
+				}
+
+				txHash = tx.getHash().toJsonString();
+			}
+
+			return txHash;
+
+		} finally {
+			LOGGER.debug("eth_sendTransaction({}): {}", args, txHash);
+		}
+	}
+
+	@Override
+	public String sendRawTransaction(String rawData) {
+		String s = null;
+		try {
+			Transaction tx = new ImmutableTransaction(stringHexToByteArray(rawData));
+
+			if (null == tx.getGasLimit() || null == tx.getGasPrice() || null == tx.getValue()) {
```

Comment:
I would take advantage of having to change this and would make the `null == something` to be `something == null`. I think it is more readable and clear since we're using Java here 😂 .

Reply:
Hi, thanks for your review.
This part of code is pretty much as it was before. I guess we will keep it for now